File: 985-sum-of-even-numbers-after-queries/985-sum-of-even-numbers-after-queries.js
/**
 * @param {number[]} nums
 * @param {number[][]} queries
 * @return {number[]}
 */
var sumEvenAfterQueries = function(nums, queries) {
    var ans = [] ;
    var even_sum = 0 ; 
    for (let num of nums) {
        if (num % 2 == 0) {
            even_sum += num ;
        }
    }
    for (let query of queries) {
        if (nums[query[1]] % 2 == 0) {
            even_sum -= nums[query[1]] ;
        }
        nums[query[1]] += query[0] ;
        if (nums[query[1]] % 2 == 0) {
            even_sum += nums[query[1]] ;
        }
        ans.push(even_sum) ;
    }
    return ans ;
};

File: 985-sum-of-even-numbers-after-queries/985-sum-of-even-numbers-after-queries.cpp
class Solution {
public:
    vector<int> sumEvenAfterQueries(vector<int>& nums, vector<vector<int>>& queries) {
        int n = nums.size() ;
        int even_sum = 0 ;
        for (int i = 0 ; i < n ; i++) {
            if (nums[i] % 2 == 0) {
                even_sum += nums[i] ;
            }
        }
        vector<int> ans ;
        for (auto &query : queries) {
            if (nums[query[1]] % 2 == 0) {
                even_sum -= nums[query[1]] ;            
            }
            nums[query[1]] += query[0] ;
            if (nums[query[1]] % 2 == 0) {
                even_sum += nums[query[1]] ;
            }
            ans.push_back(even_sum) ;
        }
        return ans ;
    }
};